cmd/pcap_converter/main.go

```diff
@@ -6,36 +6,140 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"reflect"
 	"runtime"
 	"strings"
+	"time"
 
+	"github.com/ParsePlatform/flashback"
 	"github.com/google/gopacket/pcap"
-	"github.com/mongodb/mongo-tools/common/bsonutil"
-	"github.com/mongodb/mongo-tools/common/json"
 	"github.com/tmc/mongocaputils"
 	"github.com/tmc/mongoproto"
 	"gopkg.in/mgo.v2/bson"
 )
 
 var (
-	pcapFile        = flag.String("f", "-", "pcap file (or '-' for stdin)")
+	pcapFile	= flag.String("f", "-", "pcap file (or '-' for stdin)")
+	bsonFile	= flag.String("o", "flashback.bson", "bson output file, will be overwritten")
 	packetBufSize   = flag.Int("size", 1000, "size of packet buffer used for ordering within streams")
 	continueOnError = flag.Bool("continue_on_error", false, "Continue parsing lines if an error is encountered")
+	debug           = flag.Bool("debug", false, "Print debug-level output")
 )
 
-// Converts the raw BSON doc in op.Query to extended JSON
-func rawBSONToJSON(rawBSON []byte) (interface{}, error) {
-	// Use bson.D to preserve order when unmarshalling
-	// http://godoc.org/github.com/mongodb/mongo-tools/common/bsonutil#MarshalD
-	var data bson.D
-	if err := bson.Unmarshal(rawBSON, &data); err != nil {
-		return nil, err
+type Operation struct {
+	Ns		string			`bson:"ns"`
+	Timestamp	time.Time		`bson:"ts"`
+	NToSkip		int32			`bson:"ntoskip,omitempty"`
+	NToReturn	int32			`bson:"ntoreturn,omitempty"`
+	InsertDoc	bson.D			`bson:"o,omitempty"`
+	QueryDoc	bson.D			`bson:"query,omitempty"`
+	UpdateDoc	bson.D			`bson:"updateobj,omitempty"`
+	CommandDoc	bson.D			`bson:"command,omitempty"`
+	Type		flashback.OpType	`bson:"op"`
+}
+
+func ParseQuery(opQuery []byte) (bson.D, error) {
+	var query bson.D
+	err := bson.Unmarshal(opQuery, &query)
+	return query, err
+}
+
+func (op *Operation) HandleCommand(opCommand *mongoproto.OpQuery, f *os.File) error {
+	var err error
+	op.Type = flashback.Command
+	op.CommandDoc, err = ParseQuery(opCommand.Query)
+	if err != nil {
+		return err
 	}
-	bsonAsJSON, err := bsonutil.ConvertBSONValueToJSON(data)
+	return op.Write(f)
+}
+
+func (op *Operation) HandleQuery(opQuery *mongoproto.OpQuery, f *os.File) error {
```

Comment:
If you switch to use the flashback.Op struct, you'll have to move away from these receiver methods and just use simple methods. Are these just here for convenience or is there a reason to keep using them?